src/Analyser/TypeSpecifier.php

```diff
@@ -42,126 +49,142 @@ public function __construct(\PhpParser\PrettyPrinter\Standard $printer)
+						$scope,
+						$expressions[0],
+						($context & self::CONTEXT_TRUE) ? self::CONTEXT_FALSEY : ~self::CONTEXT_FALSEY
+					);
 				} elseif ($constantName === 'true') {
-					return $this->specifyTypesInCondition($scope, $expressions[0], $negated);
+					return $this->specifyTypesInCondition(
+						$scope,
+						$expressions[0],
+						($context & self::CONTEXT_TRUE) ? self::CONTEXT_TRUTHY : ~self::CONTEXT_TRUTHY
+					);
 				}
 			}
 		} elseif ($expr instanceof Node\Expr\BinaryOp\NotEqual) {
 			return $this->specifyTypesInCondition(
 				$scope,
-				new Node\Expr\BinaryOp\Equal($expr->left, $expr->right),
-				!$negated
+				new Node\Expr\BooleanNot(new Node\Expr\BinaryOp\Equal($expr->left, $expr->right)),
+				$context
 			);
 		} elseif (
 			$expr instanceof FuncCall
 			&& $expr->name instanceof Name
 			&& isset($expr->args[0])
 		) {
 			$functionName = strtolower((string) $expr->name);
-			$expr = $expr->args[0]->value;
+			$innerExpr = $expr->args[0]->value;
 			switch ($functionName) {
 				case 'is_int':
 				case 'is_integer':
 				case 'is_long':
-					return $this->create($expr, new IntegerType(), $negated);
+					return $this->create($innerExpr, new IntegerType(), $context);
 				case 'is_float':
 				case 'is_double':
 				case 'is_real':
-					return $this->create($expr, new FloatType(), $negated);
+					return $this->create($innerExpr, new FloatType(), $context);
 				case 'is_null':
-					return $this->create($expr, new NullType(), $negated);
+					return $this->create($innerExpr, new NullType(), $context);
 				case 'is_array':
-					return $this->create($expr, new ArrayType(new MixedType()), $negated);
+					return $this->create($innerExpr, new ArrayType(new MixedType()), $context);
 				case 'is_bool':
-					return $this->create($expr, new TrueOrFalseBooleanType(), $negated);
+					return $this->create($innerExpr, new TrueOrFalseBooleanType(), $context);
 				case 'is_callable':
-					return $this->create($expr, new CallableType(), $negated);
+					return $this->create($innerExpr, new CallableType(), $context);
 				case 'is_resource':
-					return $this->create($expr, new ResourceType(), $negated);
+					return $this->create($innerExpr, new ResourceType(), $context);
 				case 'is_iterable':
-					return $this->create($expr, new IterableIterableType(new MixedType()), $negated);
+					return $this->create($innerExpr, new IterableIterableType(new MixedType()), $context);
 				case 'is_string':
-					return $this->create($expr, new StringType(), $negated);
+					return $this->create($innerExpr, new StringType(), $context);
+				case 'is_object':
+					return $this->create($innerExpr, new ObjectWithoutClassType(), $context);
 				case 'is_numeric':
-					return $this->create($expr, new UnionType([
+					return $this->create($innerExpr, new UnionType([
 						new StringType(),
 						new IntegerType(),
 						new FloatType(),
-					]), $negated);
+					]), $context);
+				case 'is_a':
```

Comment:
Add support for `$args[2]=true` please. In this case result will be `<inputType>&<resType>`.

Reply:
Can you elaborate on what you need?

Reply:
I still don't understand your comment, but I have improved the code to support the third argument.

Reply:
```php
public function createEnum(string $enumClass, string $value): BaseEnum
{
	if (!is_a($enumClass, BaseEnum::class, true)) {
		throw new LogicException("$enumClass is not correct enum class");
	}

	return $enumClass::create($value); // PHPStan should know that $enumClass is string&BaseEnum
}
```

Reply:
@pepakriz Thanks for the example, now it clear. We can't support that now, because that would require constant types (to know that the value is not just a string, but a specific string) which are not yet implemented.

Reply:
Yeah, you are right. :+1: